internal/users/tempentries/export_test.go

```diff
@@ -24,3 +29,11 @@ func (r *PreAuthUserRecords) DeletePreAuthUser(uid uint32) {
 
 	r.deletePreAuthUser(uid)
 }
+
+func OverrideMaxTemporaryUserLifetime(t *testing.T, lifetime time.Duration) {
```

Comment:
> users/preauth: Garbage collect unused pre-registered users after a timeout

This commit doesn't really belong on this PR, does it?

Also, does it really effectively thwart DoS attacks? The attacker could just do another 4096 SSH requests after 15 minutes, filling up the max pre-auth users again. If it's that easy to circumvent, I'm not sure it's worth doing it at all. I think a DoS is something the admins should look into, so printing an error and letting the admins restart authd to reset the max pre-auth users seems like an ok solution to me.